### ISSUE TITLE
screen freezes on ios  and no touch support on Layout with entering or exiting animation

### ISSUE THREAD

**Issue by geoapostolidis** [software-mansion/react-native-reanimated#3388]
## Description

On ios the screen freezes on Layout with entering or exiting animation. When i navigate to the page that has layout on navigate 
back then the screen freezes and  has no touch support. The problem is that there is no error in console in order to debug it.

## Package versions

i have these packages installed 

"@react-navigation/native": "^6.0.11",
"@react-navigation/stack": "^6.2.2",
"react": "18.0.0",
"react-native": "0.69.1",
"react-native-reanimated": "^2.9.1"

## Affected platforms

- [ ] Android
- [x] iOS
- [ ] Web

**Comment by ddikodroid** [comment]
related to https://github.com/software-mansion/react-native-reanimated/issues/3124

**Comment by Tauka** [comment]
Experiencing same issue

**Comment by anhnch** [comment]
For me, all I did is just an import Animated from 'react-native-reanimated'. When I go to a screen and go back, the screen is not responding to touch/scroll. The screen is still functioning properly, but you simply can't do anything. No error, no log, nothing. Because the problem occurs when navigating and react-navigation does use react-native-reanimated, so I think it's incompatibility with react-navigation

react-native-reanimated: 2.9.1
@react-navigation/drawer: 6.4.3
@react-navigation/native: 6.0.11
@react-navigation/stack: 6.2.2
react-native-screens: 3.15.0

**Comment by Tauka** [comment]
Removing `GestureHandler` from `react-native-gesture-handler` fixed freezes for me. Read it from here: https://github.com/software-mansion/react-native-reanimated/issues/3331

**Comment by amin79** [comment]
> For me, all I did is just an import Animated from 'react-native-reanimated'. When I go to a screen and go back, the screen is not responding to touch/scroll. The screen is still functioning properly, but you simply can't do anything. No error, no log, nothing. Because the problem occurs when navigating and react-navigation does use react-native-reanimated, so I think it's incompatibility with react-navigation
> 
> react-native-reanimated: 2.9.1 @react-navigation/drawer: 6.4.3 @react-navigation/native: 6.0.11 @react-navigation/stack: 6.2.2 react-native-screens: 3.15.0

Same problem here. Did you find any solutions?

**Comment by AhmadHoranieh** [comment]
@Tauka Can you explain what you did, please?

**Comment by entangledloops** [comment]
Is there a workaround or ETA on a fix for this?

**Comment by lurdharry** [comment]
Thanks so much for creating this issue, it took me a while to even know what’s causing the freeze.

**Comment by aLemonFox** [comment]
> @AhmadHoranieh Instead of using `GestureHandler` from `react-native-gesture-handler`, I used `PanGestureHandler`

This works for me too, but it's not ideal since PanGestureHandler is using the old API which is no longer maintained. Any news on this issue?

**Comment by vanenshi** [comment]
I am facing this issue too

**Comment by Lakston** [comment]
Same problem I just spent half a day trying to figure this out.

reanimated v2.14.0
react-native v0.70.5

and just using fades:

```
          <Animated.View
                  key={"uniqueKey"}
                  entering={FadeIn.duration(300)}
                  exiting={FadeOut.duration(300)}
                >
                  {...}
                </Animated.View>
```

As soo as I hit the back button the layout is frozen and I need to reload the app

**Comment by lewisd1996** [comment]
Also experiencing this problem, after navigating to a screen, the screen generally works fine but the gesture handler is frozen and requires a hot reload to kick in. Unsure if this happens in production.

I am using `GestureDetector` with `Gesture.Pan()`

**Comment by pelayomartinez** [comment]
Also experiencing this issue - not even using react-native-gesture-handler. Screen becomes unresponsive to touch after the screen containing the layout transitions is popped from stack navigator

**Comment by kirbi96** [comment]
resolve for me after update to 3.2.0 version

**Comment by Lakston** [comment]
do you mean 3.0.2 ?

I tried to update, but upgrading reanimated breaks the async storage lib apparently, see t[his issue](https://github.com/software-mansion/react-native-reanimated/issues/4267), so I'm stuck, cant use the pre made animations on 2.X and can not upgrade to 3.X because it breaks our app.

**Comment by t0ma5h** [comment]
Having the same issues weirdly after installing firebase as a dependency.

**Comment by filippobarcellos** [comment]
@t0ma5h Did you manage to get it fixed? I'm having the same problem after adding firebase as well. Not sure how they are related though.

**Comment by qaws5503** [comment]
Same problem here. Does someone solved this problem?

**Comment by MacPiston** [comment]
Recently encountered this issue with reanimated v2. I can confirm that with reanimated 3.1.0, @react-navigation/native 6.1.6 and @react-navigation/stack 6.3.16. I can't relate to AsyncStorage issue as I haven't tested that, but even if that was the case having responsive screens again is much more important.

**Comment by t0ma5h** [comment]
@filippobarcellos yeah, upgraded reanimated to 3.1.0. Be sure to reset the metro cache after you update 👌

**Comment by ddikodroid** [comment]
> > Recently encountered this issue with reanimated v2. I can confirm that with reanimated 3.1.0, @react-navigation/native 6.1.6 and @react-navigation/stack 6.3.16 screens are responsive again. I can't relate to AsyncStorage issue as I haven't tested that, but even if that was the case having responsive screens again is much more important.
> 
> Well since not being able to access the async storage completely breaks our app, it feels more important to have a fix for that first !

why don't you use mmkv instead?

**Comment by Ashalbulk** [comment]
looks like that the reason is in gesture, try to use
working for me on android using react-navigation and the gesture handler HOC

`AppRegistry.registerComponent(appName, () => gestureHandlerRootHOC(appRoot));`

instead of <GestureHandlerRootView>

**Comment by patrycjakalinska** [comment]
hi I've tested this issue on [Combined example](https://github.com/software-mansion/react-native-reanimated/blob/main/apps/common-app/src/apps/reanimated/examples/LayoutAnimations/Combined.tsx) from our App that's similar to your repro (on RN `0.80-rc.3` and Reanimated `3.18.0`) and there was no crash. Try to run your app on latest Reanimated and let me know if this helps (also try to create clean RN app)
